modules/src/core/ics04_channel/context.rs

```diff
@@ -226,15 +226,10 @@ pub trait ChannelKeeper {
                 )?;
             }
             PacketResult::Ack(res) => {
-                match res.seq_number {
-                    Some(s) => {
-                        //Ordered Channel
-                        self.store_next_sequence_ack(res.port_id, res.channel_id, s)?;
-                    }
-                    None => {
-                        //Unordered Channel
-                        self.delete_packet_commitment(&res.port_id, &res.channel_id, res.seq)?;
-                    }
+                self.delete_packet_commitment(&res.port_id, &res.channel_id, res.seq)?;
```

Comment:
This is the actual code that fixes the bug and was added with the last commit (de79898f62bfe26001a26ed21e8656829578ec17). All other changes are just to enable testing. Run the `core::ics26_routing::handler::tests::routing_module_and_keepers` test *sans* de79898f62bfe26001a26ed21e8656829578ec17 to see the failing test.